### PR TITLE
H265 Support in WebRTC Canary

### PR DESCRIPTION
**Pull request by niyatim23**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: canary/webrtc-c/jobs/orchestrator.groovy [hunk 1]
```diff
@@ -188,6 +188,75 @@ pipeline {
...
+                                booleanParam(name: 'USE_TURN', value: true),
+                                booleanParam(name: 'TRICKLE_ICE', value: true),
+                                booleanParam(name: 'USE_IOT', value: true),
+                                booleanParam(name: 'USE_MBEDTLS', value: false),
+                                string(name: 'DURATION_IN_SECONDS', value: LONG_RUNNING_DURATION_IN_SECONDS.toString()),
+                                string(name: 'VIDEO_CODEC', value: "h265"),
+                                string(name: 'MASTER_NODE_LABEL', value: "openssl-master"),
+                                string(name: 'VIEWER_NODE_LABEL', value: "openssl-viewer"),
+                                string(name: 'RUNNER_LABEL', value: "WebrtcLongRunningOpenSSL-H265"),
+                                string(name: 'SCENARIO_LABEL', value: "WebrtcLongRunning"),
```
Comment (disa6302) [inline canary/webrtc-c/jobs/orchestrator.groovy:221]: WebrtcLongRunning-H265

File: canary/webrtc-c/jobs/orchestrator.groovy [hunk 2]
```diff
@@ -188,6 +188,75 @@ pipeline {
...
+                                booleanParam(name: 'USE_TURN', value: true),
+                                booleanParam(name: 'TRICKLE_ICE', value: true),
+                                booleanParam(name: 'USE_IOT', value: false),
+                                booleanParam(name: 'USE_MBEDTLS', value: true),
+                                string(name: 'DURATION_IN_SECONDS', value: PERIODIC_DURATION_IN_SECONDS.toString()),
+                                string(name: 'VIDEO_CODEC', value: "h265"),
+                                string(name: 'MASTER_NODE_LABEL', value: "mbedtls-master"),
+                                string(name: 'VIEWER_NODE_LABEL', value: "mbedtls-viewer"),
+                                string(name: 'RUNNER_LABEL', value: "WebrtcPeriodicStaticMbedTLS-H265"),
+                                string(name: 'SCENARIO_LABEL', value: "MbedTLSPeriodic"),
```
Comment (disa6302) [inline canary/webrtc-c/jobs/orchestrator.groovy:238]: MbedTLSPeriodic-H265

File: canary/webrtc-c/jobs/orchestrator.groovy [hunk 3]
```diff
@@ -188,6 +188,75 @@ pipeline {
...
+                                booleanParam(name: 'USE_TURN', value: true),
+                                booleanParam(name: 'TRICKLE_ICE', value: true),
+                                booleanParam(name: 'USE_IOT', value: false),
+                                booleanParam(name: 'USE_MBEDTLS', value: true),
+                                string(name: 'DURATION_IN_SECONDS', value: LONG_RUNNING_DURATION_IN_SECONDS.toString()),
+                                string(name: 'VIDEO_CODEC', value: "h265"),
+                                string(name: 'MASTER_NODE_LABEL', value: "mbedtls-master"),
+                                string(name: 'VIEWER_NODE_LABEL', value: "mbedtls-viewer"),
+                                string(name: 'RUNNER_LABEL', value: "WebrtcLongRunningStaticMbedTLS-H265"),
+                                string(name: 'SCENARIO_LABEL', value: "WebrtcLongRunning"),
```
Comment (disa6302) [inline canary/webrtc-c/jobs/orchestrator.groovy:255]: WebrtcLongRunning-H265

File: canary/webrtc-c/scripts/v_init.sh [hunk 4]
```diff
@@ -7,10 +7,11 @@ export CANARY_IS_MASTER=FALSE  # if you are running the app as master
 export CANARY_TRICKLE_ICE=TRUE  # If you would like to enable trickle ICE
 export CANARY_USE_TURN=TRUE  # This means turn urls are are also selected as an alternate path
 export CANARY_DURATION_IN_SECONDS=90 # How long you want to run this in seconds.
-export CANARY_FORCE_TURN=FALSE #This will enforce TURN, which is your use case
-export CANARY_USE_IOT_PROVIDER=FALSE #This will enforce use of IoT credential provider
+export CANARY_FORCE_TURN=FALSE # This will enforce TURN, which is your use case
+export CANARY_USE_IOT_PROVIDER=FALSE # This will enforce use of IoT credential provider
 export CANARY_LABEL=Profiling
 export CANARY_IS_PROFILING_MODE=TRUE
+export CANARY_VIDEO_CODEC=h265 # This sets the video codec (can take h264(default), h265)
```
Comment (disa6302) [inline canary/webrtc-c/scripts/v_init.sh:14]: Make it consistent with the init script

File: canary/webrtc-c/src/CanaryWebrtc.cpp [hunk 5]
```diff
@@ -269,19 +268,25 @@ VOID sendCustomFrames(Canary::PPeer pPeer, MEDIA_STREAM_TRACK_KIND kind, UINT64
...
+    UINT32 minFrameSize = CANARY_METADATA_SIZE + ((dataRate / 8) / frameRate);
+    UINT32 maxFrameSize = (CANARY_METADATA_SIZE + ((dataRate / 8) / frameRate)) * 2;
     PBYTE canaryFrameData = NULL;
-    canaryFrameData = (PBYTE) MEMALLOC(actualFrameSize);
+    
+    canaryFrameData = (PBYTE) MEMALLOC(maxFrameSize);
 
     // We allocate a bigger buffer to accomodate the hex encoded string
-    frame.frameData = (PBYTE) MEMALLOC(frameSizeWithoutNalu * 2 + 1 + ANNEX_B_NALU_SIZE);
+    frame.frameData = (PBYTE) MEMALLOC(maxFrameSize * 3 + 1 + ANNEX_B_NALU_SIZE);
```
Comment (disa6302) [inline canary/webrtc-c/src/CanaryWebrtc.cpp:278]: Did you test H264 with this?

Comment (niyatim23) [reply canary/webrtc-c/src/CanaryWebrtc.cpp:278]: Yes, I ran the 90 seconds test with these changes and it works for H264 and H265 both

File: canary/webrtc-c/jobs/orchestrator.groovy
Comment (disa6302) [inline canary/webrtc-c/jobs/orchestrator.groovy:1]: Set GIT_HASH to your demos branch for testing. We need to change it back to master before merging.

File: canary/webrtc-c/jobs/orchestrator.groovy [hunk 6]
```diff
@@ -188,6 +188,75 @@ pipeline {
...
+                        build(
+                            job: NEXT_AVAILABLE_RUNNER,
+                            parameters: COMMON_PARAMS + [
+                                booleanParam(name: 'USE_TURN', value: true),
+                                booleanParam(name: 'TRICKLE_ICE', value: true),
+                                booleanParam(name: 'USE_IOT', value: false),
+                                booleanParam(name: 'USE_MBEDTLS', value: true),
+                                string(name: 'DURATION_IN_SECONDS', value: LONG_RUNNING_DURATION_IN_SECONDS.toString()),
+                                string(name: 'VIDEO_CODEC', value: "h265"),
+                                string(name: 'MASTER_NODE_LABEL', value: "mbedtls-master"),
```
Comment (disa6302) [inline canary/webrtc-c/jobs/orchestrator.groovy:252]: Set to mbedtls-h265-master

File: canary/webrtc-c/jobs/orchestrator.groovy [hunk 7]
```diff
@@ -188,6 +188,75 @@ pipeline {
...
+                                string(name: 'VIEWER_NODE_LABEL', value: "mbedtls-viewer"),
```
Comment (disa6302) [inline canary/webrtc-c/jobs/orchestrator.groovy:253]: Set to mbedtls-h265-viewer

File: canary/webrtc-c/jobs/orchestrator.groovy [hunk 8]
```diff
@@ -188,6 +188,75 @@ pipeline {
...
+                        build(
+                            job: NEXT_AVAILABLE_RUNNER,
+                            parameters: COMMON_PARAMS + [
+                                booleanParam(name: 'USE_TURN', value: true),
+                                booleanParam(name: 'TRICKLE_ICE', value: true),
+                                booleanParam(name: 'USE_IOT', value: false),
+                                booleanParam(name: 'USE_MBEDTLS', value: true),
+                                string(name: 'DURATION_IN_SECONDS', value: PERIODIC_DURATION_IN_SECONDS.toString()),
+                                string(name: 'VIDEO_CODEC', value: "h265"),
+                                string(name: 'MASTER_NODE_LABEL', value: "mbedtls-master"),
```
Comment (disa6302) [inline canary/webrtc-c/jobs/orchestrator.groovy:235]: Same here: mbedtls-h265-master and mbedtls-h265-viewer

File: canary/webrtc-c/jobs/orchestrator.groovy [hunk 9]
```diff
@@ -188,6 +188,75 @@ pipeline {
...
+                        build(
+                            job: NEXT_AVAILABLE_RUNNER,
+                            parameters: COMMON_PARAMS + [
+                                booleanParam(name: 'USE_TURN', value: true),
+                                booleanParam(name: 'TRICKLE_ICE', value: true),
+                                booleanParam(name: 'USE_IOT', value: true),
+                                booleanParam(name: 'USE_MBEDTLS', value: false),
+                                string(name: 'DURATION_IN_SECONDS', value: LONG_RUNNING_DURATION_IN_SECONDS.toString()),
+                                string(name: 'VIDEO_CODEC', value: "h265"),
+                                string(name: 'MASTER_NODE_LABEL', value: "openssl-master"),
```
Comment (disa6302) [inline canary/webrtc-c/jobs/orchestrator.groovy:218]: Set to openssl-h265-master and openssl-h265-viewer. Similarly above.

File: canary/webrtc-c/jobs/orchestrator.groovy [hunk 10]
```diff
@@ -13,9 +13,10 @@ STORAGE_EXTENDED_DURATION_IN_SECONDS = 43200 // 12 hr
 MIN_RETRY_DELAY_IN_SECONDS = 60
 COLD_STARTUP_DELAY_IN_SECONDS = 60 * 60
 GIT_URL = 'https://github.com/aws-samples/amazon-kinesis-video-streams-demos.git'
-GIT_HASH = 'master'
+GIT_HASH = 'h265-support'
```
Comment (disa6302) [inline canary/webrtc-c/jobs/orchestrator.groovy:16]: Switch back to master before merging.

Comment (niyatim23) [reply canary/webrtc-c/jobs/orchestrator.groovy]: Done!

**Pull request merged**